dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -159,5 +179,146 @@ public void testGetRelatedContentFromMultilingualContent()
+
+            final ESIndexAPI esIndexAPI = Mockito.mock(ESIndexAPI.class);
+
+            Mockito.when(esIndexAPI.isClusterInReadOnlyMode()).thenReturn(true);
+
+            assertFalse(
+                    new ESContentletAPIImpl().isCheckInSafe(contentletRelationship, esIndexAPI));
+
+        }finally{
+            if (contentType != null && contentType.id() != null){
+                contentTypeAPI.delete(contentType);
+            }
+        }
+    }
+
+    @Test
+    public void testIsCheckInSafeWithRelationshipsFieldsShouldReturnTrue()
+            throws DotDataException, DotSecurityException {
+        final long time = System.currentTimeMillis();
+        ContentType contentType = null;
+
+        try {
+            contentType = createContentType("test" + time);
+
+            final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+            final ContentletRelationships contentletRelationship = new ContentletRelationships(
+                    contentlet);
+
+            Field field = FieldBuilder.builder(RelationshipField.class).name("newRel")
+                    .contentTypeId(contentType.id()).values(String.valueOf(RELATIONSHIP_CARDINALITY.ONE_TO_MANY.ordinal()))
+                    .relationType(contentType.variable()).build();
+
+            field = fieldAPI.save(field, user);
+
+            final String fullFieldVar = contentType.variable() + StringPool.PERIOD + field.variable();
+
+            final Relationship relationship = relationshipAPI.byTypeValue(fullFieldVar);
+
+            final ContentletRelationshipRecords relationshipsRecord = contentletRelationship.new ContentletRelationshipRecords(
+                    relationship,false);
+
+            contentletRelationship
+                    .setRelationshipsRecords(CollectionsUtils.list(relationshipsRecord));
+
+            final ESIndexAPI esIndexAPI = Mockito.mock(ESIndexAPI.class);
+
+            Mockito.when(esIndexAPI.isClusterInReadOnlyMode()).thenReturn(true);
+
+            assertTrue(
+                    new ESContentletAPIImpl().isCheckInSafe(contentletRelationship, esIndexAPI));
+
+        }finally{
+            if (contentType != null && contentType.id() != null){
+                contentTypeAPI.delete(contentType);
+            }
+        }
+    }
+
+    @Test
+    public void testIsCheckInSafeWithoutRelationshipsShouldReturnTrue() {
+        final ESIndexAPI esIndexAPI = Mockito.mock(ESIndexAPI.class);
+
+        Mockito.when(esIndexAPI.isClusterInReadOnlyMode()).thenReturn(true);
+
+        assertTrue(
+                new ESContentletAPIImpl().isCheckInSafe(null, esIndexAPI));
+    }
+
+
+    private ContentType createContentType(final String name) throws DotSecurityException, DotDataException {
+        return contentTypeAPI.save(ContentTypeBuilder.builder(SimpleContentType.class).folder(
+                FolderAPI.SYSTEM_FOLDER).host(Host.SYSTEM_HOST).name(name)
+                .owner(user.getUserId()).build());
+    }
 
```

Comment:
I think are we missing a test: testCheckInWithoutLegacyRelationshipsAndReadOnlyClusterShouldThrowAnException?

Reply:
For this case we shouldn't throw an exception